quarkus/runtime/src/main/java/org/keycloak/quarkus/runtime/configuration/mappers/PropertyMapper.java

```diff
@@ -349,16 +352,30 @@ public Builder<T> isEnabled(BooleanSupplier isEnabled) {
         /**
          * Set the validator, overwriting the current one.
          */
-        public Builder<T> validator(BiConsumer<PropertyMapper<T>, ConfigValue> validator) {
-            this.validator = validator;
+        public Builder<T> validator(Consumer<String> validator) {
+            this.validator = (mapper, value) -> mapper.validateExpectedValues(value,
+                    (c, v) -> validator.accept(v));
+            if (!Objects.equals(this.description, this.option.getDescription())) {
+                throw new AssertionError("appendValidateEnabled was already called");
+            }
```

Comment:
Might seem confusing for the `validator()` method, comparing some descriptions and having an unclear relationship with the `appendValidateEnabled`. 

I'm not a big fan of the `appendValidateEnabled` either as I find it confusing as well - like having the same functionality as `isEnabled()` except the aspect the prop will be shown in help. I'd still rather extend the `isEnabled()` signature about some flag for showing it in help and removing the `appendValidateEnabled` at all. 

@shawkins WDYT?

Reply:
> Might seem confusing for the validator() method, comparing some descriptions and having an unclear relationship with the appendValidateEnabled.

The message can be exanded to reference the description if you like.

> @shawkins WDYT?

I disagree. There's a good bit of complexity with how the isEnabled logic is consumed and it won't help to add another check for an additional flag. Keeping this as stictly validation related encapsulates the complexity to the PropertyMapper / builder. And if you end up setting a validator based upon an additional flag you'll have the same issue of interdependency between calls.

Reply:
> The message can be exanded to reference the description if you like.

Would be great - like when the `AssertionError` is thrown with the `appendValidateEnabled was already called` msg, I don't know what's the problem. "I'm not able to call appendValidateEnabled more times? Was the validator() called twice? Does the order of the builder method matter? ...."

Reply:
> I disagree. There's a good bit of complexity with how the isEnabled logic is consumed and it won't help to add another check for an additional flag. Keeping this as stictly validation related encapsulates the complexity to the PropertyMapper / builder. And if you end up setting a validator based upon an additional flag you'll have the same issue of interdependency between calls.

As we're discussing more DevEx as users are not affected by this, it might be sufficient to improve the error message as the Javadoc for the `appendValidateEnabled` should be sufficient. 

So, ok, not a big deal for me :)